creation/interfaces-and-networks.md

```diff
@@ -578,13 +578,22 @@ not automatic; it should be configured via cloud init, as shown below:
         pod: {}
       volumes:
       - cloudInitNoCloud:
+          networkData: |
+            version: 2
+            ethernets:
+              eth0:
+                addresses: [ fd10:0:2::2/120 ]
+                dhcp4: true
+                gateway6: fd10:0:2::1
+                nameservers:
+                  addresses: [ <i>dns_server_ip</i> ]
+                  search:
+                    - default.svc.cluster.local
```

Comment:
This seems reasonable, but was not in the original script. Was it part of /etc/resolv.conf?

Reply:
the original scripts appends stuff to `resolv.conf`.

This was already there.

Reply:
Issue is that the "default" is taken from the Pod/VM namespace.

Reply:
Which is the implicit namespace used in the example; it is correct, if only a bit implicit. And implicit is bad. 

Would you rather have me create the VM in a different namespace ? Would that make it more explicit ?

Or use a  *namespace_name* place-holder, as I'm doing w/ the **dns_server_ip** ?

Reply:
> the original scripts appends stuff to resolv.conf.

Who populates it though? How does guest know what to set there if there is no DHCPv6 server?

> Would you rather have me create the VM in a different namespace ? Would that make it more explicit ?

User does not usually need to worry about namespaces. They just create VMs/Pod and those get created in user's own namespace. So requiring users to specify the namespace here is not only one extra step (copy the namespace attribute to cloudInit), it also requires them to know/worry what is the name of their namespace. I think we should avoid that. 

If we omit the `search` section, is it then taken from resolv.conf (if it indeed is in the original resolv.conf)?

Since we will eventually add a method that pushes IP and DNS info to the guest, we can treat this as a temporary state and require users to worry about their namespace.

Reply:
> > the original scripts appends stuff to resolv.conf.
> 
> Who populates it though? How does guest know what to set there if there is no DHCPv6 server?

But there *is* a DHCP(v4) server - we just have to 'ask' for dhcpv4 enabled.

I now understand more than I did before, thanks for this question.

> 
> > Would you rather have me create the VM in a different namespace ? Would that make it more explicit ?
> 
> User does not usually need to worry about namespaces. They just create VMs/Pod and those get created in user's own namespace. So requiring users to specify the namespace here is not only one extra step (copy the namespace attribute to cloudInit), it also requires them to know/worry what is the name of their namespace. I think we should avoid that.
> 
> If we omit the `search` section, is it then taken from resolv.conf (if it indeed is in the original resolv.conf)?

No, if you omit it, `resolv.conf` will be generated without it, which leads to DNS not working.

> 
> Since we will eventually add a method that pushes IP and DNS info to the guest, we can treat this as a temporary state and require users to worry about their namespace.

Long story short, and this relates to your other question about dhcpv4:
  - we can omit dhcpv4: that **requires** us to specify the 'search domains'
  - we enable dhcpv4: that allows us to omit the 'search domains'

It can go either way. Your call ? (I vote for having dhcpv4 explicitly set).

Reply:
If dhcp4 gives us search, I vote for it too. And omit domains. It is implicit, but for user's comfort (am I hypocrite?)

Reply:
At least use "template" for the namespace just as you do for DNS.

<i>pod_namespace</i>

I hope cloudInit explodes if a user blindly copy-pastes this.